openerp/addons/base/res/ir_property.py

```diff
@@ -71,8 +56,8 @@ def _update_values(self, cr, uid, ids, values):
         prop = None
         type_ = values.get('type')
         if not type_:
-            if ids:
-                prop = self.browse(cr, uid, ids[0])
+            if self.exists():
+                prop = self[0]
```

Comment:
Beware: `self.exists()` returns the subset of `self` that exists; it does not guarantee that all records in `self` exist!
The test `if ids` (`ids` is a nonempty list) should be migrated to `if self` (`self` is a nonempty recordset) in the new API.